app/assets/javascripts/sufia/relationships/table.es6

```diff
@@ -0,0 +1,206 @@
+import { RelationshipsTableRow } from './table_row'
+
+export class RelationshipsTable {
+
+  /**
+   * Initializes the class in the context of an individual table element
+   * @param {jQuery} element the table element that this class represents
+   */
+  constructor(element) {
+    this.$element = element;
+    this.form_action = this.$element.parents("form").attr("action");
+    this.query_url = this.$element.data('query-url');
+    this.existing_related_works_values = this.$element.find("input.related_works_ids:not(.new-form-control)").map(function(i, e){ return e.value; });
+
+    // TODO: Fall back to just cloning rows and removing rows for form posting
+    if (!this.query_url)
+      return;
+
+    this.bindAddButton();
+    this.bindRemoveButton();
+    this.bindKeyEvents();
+  }
+
+  /**
+   * Handle click events by the "Add" button in the table, setting a warning
+   * message if the input is empty or calling the server to handle the request
+   */
+  bindAddButton() {
+    let $this = this;
+
+    $this.$element.on("click", ".btn-add-row", function(event) {
+      let $row = $(this).parents("tr:first");
+      let $input = $row.find("input.new-form-control");
+
+      // Display an error when the input field is empty, or if the work ID is already related,
+      // otherwise clone the row and set appropriate styles
+      if ($input.val() == "") {
+        $this.setWarningMessage($row, "ID cannot be empty.");
+      } else if ($.inArray($input.val(), $this.existing_related_works_values) > -1) {
+        $this.setWarningMessage($row, "Work is already related.");
+      } else {
+
+        let query_url = $this.query_url.replace('$id', $input.val());
+        $this.hideWarningMessage($row);
+        $this.callAjax({
+          row: $row,
+          table: $this.$element,
+          input: $input,
+          url: $this.form_action,
+          query_url: query_url,
+          on_error: $this.handleError,
+          on_success: $this.handleAddRowSuccess
+        });
+      }
+    });
+  }
+
+  /**
+   * Find and fire off the click event for the "Add" button
+   * @param {jQuery} $row the row containing the add button to click
+   */
+  clickAddbutton($row) {
+    $row.find(".btn-add-row").click();
+  }
+
+  /**
+   * Handle click events by the "Remove" buttons in the table, and calling the
+   * server to handle the request
+   */
+  bindRemoveButton() {
+    let $this = this;
```

Comment:
I need both reference to the RelationshipsTable class instance, and the button which fired the click event. The double arrow would give reference to the button, but I would still need reference to the RelationshipsTable class instance to `.callAjax`, right? 

These references are passed to the error and success callbacks so that they can do things like setting the warning message, etc.